server/src/main/java/org/apache/druid/server/initialization/jetty/JettyServerInitUtils.java

```diff
@@ -122,4 +127,25 @@ public static void addAllowHttpMethodsFilter(ServletContextHandler root, List<St
         null
     );
   }
+  
+  public static void maybeAddHSTSPatternRule(ServerConfig serverConfig, RewriteHandler rewriteHandler)
+  {
+    if (serverConfig.isEnableHSTS()) {
+      rewriteHandler.addRule(getHSTSHeaderPattern());
+    }
+  }
+
+  public static void maybeAddHSTSRewriteHandler(ServerConfig serverConfig, HandlerList handlerList)
+  {
+    if (serverConfig.isEnableHSTS()) {
+      RewriteHandler rewriteHandler = new RewriteHandler();
+      rewriteHandler.addRule(getHSTSHeaderPattern());
+      handlerList.addHandler(rewriteHandler);
+    }
+  }
+
+  private static HeaderPatternRule getHSTSHeaderPattern()
+  {
+    return new HeaderPatternRule("*", "Strict-Transport-Security", "max-age=63072000; includeSubDomains");
```

Comment:
should this be configurable?

Reply:
+1. We can keep this as default and add a new config param which overrides this.

Reply:
this is a standard header value. If there is a need for it, we can add a new config option later on. I would like to avoid adding another config param right now. If users ask for it, adding a new config option is easy.